### PR TITLE
Allow links to JUnit Jupiter Javadoc, add Gradle version catalog

### PR DESCRIPTION
**Pull request by scordio**
This solves the gap mentioned at https://github.com/mockito/mockito/pull/3230#issuecomment-1890393656.

`javadoc` can create external links to libraries documented with "mismatching" modularity, like for JUnit Jupiter, which offers modular documentation. However, a warning is generated when the mismatch happens, and this is a problem when using the `-Werror` option to treat warnings as errors.

To circumvent this problem without dropping the `-Werror` option, Java 18 added the

`--link-modularity-mismatch=(warn|info)`

command line option to specify how external documentation with wrong modularity should be reported.

See: https://bugs.openjdk.org/browse/JDK-8274639

In addition, this introduces a [`libs.versions.toml`](https://docs.gradle.org/current/userguide/platforms.html#sub:conventional-dependencies-toml) version catalog, suggested at #3384.

### CODE REVIEW

**Comment by scordio** [comment]
> All the CI jobs with Java < 18 should skip the Javadoc generation. I'll check how this could be done.

This is resolved with a conditional configuration of the `javadoc` task.

File: gradle/mockito-junit-jupiter/javadoc.gradle [hunk 1]
```diff
@@ -34,6 +34,12 @@ tasks.named("javadoc", Javadoc) {
     options.use = true
     options.windowTitle = "Mockito JUnit Jupiter ${project.version} API"
 
+    // https://bugs.openjdk.org/browse/JDK-8274639
+    if (JavaVersion.current() >= JavaVersion.VERSION_18) {
+        options.addStringOption('-link-modularity-mismatch', 'info')
+        options.links("https://junit.org/junit5/docs/5.11.0/api/")
```
Comment (scordio) [inline gradle/mockito-junit-jupiter/javadoc.gradle:40]: Ideally, the JUnit version number shouldn't be hardcoded.

I see it appears here:

https://github.com/mockito/mockito/blob/f6450a01f27d0e4e7e95a52025b000726558774d/gradle/dependencies.gradle#L8

What would be the proper way to expose that value to this file?

Comment (TimvdLippe) [reply gradle/mockito-junit-jupiter/javadoc.gradle:40]: If I understand correctly, we need to define the `versions` to be exposed on the project as well. For that, we can do the following:

```groovy
ext {
    libraries = [:]
    libraryVersions = [:]
}

libraryVersions.bytebuddy = '1.15.0'
libraryVersions.junitJupiter = '5.11.0'
libraryVersions.errorprone = '2.23.0'
```

And then here this should work:

```groovy
options.links("https://junit.org/junit5/docs/${libraryVersions.junitJupiter}/api/")
```

Comment (scordio) [reply gradle/mockito-junit-jupiter/javadoc.gradle]: Solved with the introduction of [`libs.versions.toml`](https://docs.gradle.org/current/userguide/platforms.html#sub:conventional-dependencies-toml).

**Review by TimvdLippe — APPROVED** [review]
Nice fix, thanks for the contribution!

**Pull request merged**